api/runnerpool/placer_tracker.go

```diff
@@ -113,7 +113,24 @@ func (tr *placerTracker) HandleDone() {
 
 // RetryAllBackoff blocks until it is time to try the runner list again. Returns
 // false if the placer should stop trying.
-func (tr *placerTracker) RetryAllBackoff(numOfRunners int) bool {
+func (tr *placerTracker) RetryAllBackoff(numOfRunners int, err error) bool {
+
+	// If the last call to obtain runners failed due to a user error or
+	// misconfiguration then do not re-attempt to obtain a runner.
+	if err != nil {
```

Comment:
The only other thing here I'm not sure on is whether this should be guarded by `numOfRunners == 0` as well as this check.

Open to second opinions here: @ostrain ?

Reply:
Jan and I discussed the advantages/disadvantages of adding this clause:

(+)
1. It is a tighter more defensive control over this 'fail-fast' mechanism.
2. If the user had existing runners provisioned before miss-configuring the platform, then, they may have some time to correct the error before their invocation attempts being to fail. So it is more robust.

(-)
1. When (re-)configuring the platform incorrectly, the user may not immediately notice their mistake. Failures will only occur if/when the runners are re-cycled or require an increase capacity.

I have made the change, but, can roll it back if @ostrain has more info.